src/store/modules/user/actions.ts

```diff
@@ -222,6 +225,101 @@ const actions: ActionTree<UserState, RootState> = {
       console.error(error);
     }
     return []
+  },
+
+  async setProductIdentificationPref({ commit, state }, payload) {
+    let prefValue = JSON.parse(JSON.stringify(state.productIdentificationPref))
+    const eComStoreId = (state.currentEComStore as any).productStoreId
+
+    const fromDate = UserService.getProductIdentificationPrefFromDate(eComStoreId)
```

Comment:
We could use getProductIdentificationPref

Reply:
Moved the code to get fromDate in action itself.